vochain/state_trees.go

```diff
@@ -161,3 +198,116 @@ func StateParentChildTreeCfg(parent, child string, key []byte) (statedb.TreeConf
 	childTree := StateChildTreeCfg(child)
 	return parentTree, childTree.WithKey(key)
 }
+
+// rootLeafGetRoot is the GetRootFn function for a leaf that is the root
+// itself.
+func rootLeafGetRoot(value []byte) ([]byte, error) {
```

Comment:
In state_votes you made this functions methods of the state  (i.e `func (v *State) voteCountInc()`) but here are not State methods. Do this follow any specific logic?

Same for the functions below.
Also I would like to have `func (s *State)` rather than `func (v *State)`, looks nicer :)

Reply:
In general methods are used when the logic requires access to the struct object state. Functions are used when the logic can be run standalone and no state is required. I don't think abusing from methods is a good idiomatic strategy.

What I tried to do on this minor refactor is to group the methods and functions by functionality. So in this case, everything related to "state merkle trees" is moved to "state_trees.go".

Reply:
Regarding the renaming of s/v I agree but this is not related to this commit (you can field a new issue if you want).

In any case major refactors on naming and structure must be perform in some moment (such as move all the State methods to a separated package).